soundfile.py

```diff
@@ -167,6 +167,33 @@
     __libsndfile_version__ = __libsndfile_version__[len('libsndfile-'):]
 
 
+class SoundFileError(RuntimeError):
+    """Base class for all SoundFile-specific errors.
+
+    :ivar code: libsndfile internal error number or `None`
+    :ivar error_string: raw libsndfile error message or `None`
+    """
+    def __init__(self, message, code=None, error_string=None):
+        self.args = (message, code, error_string)
```

Comment:
AFAICT, this is not necessary, `args` is set automatically.

Reply:
Would be if I called base class constructor. I have changed mind here multiple times and there were probably some good reasons in between ;-). I'll think about it and maybe call the base class constructor instead.

Reply:
It might depend on the Python version, but for me this works without calling the base class constructor:

```python
>>> class MyError(RuntimeError):
...     def __init__(self, one, two=None):
...         self.one = one
...         self.two = two
... 
>>> e1 = MyError('test')
>>> e1.args
('test',)
>>> e2 = MyError('hello', 'world')
>>> e2.args
('hello', 'world') 
```

Does this not work for you?

Reply:
Actually I have not tried this, as I would never thought this can work. Every day one can learn something :-)
Thanks.

Reply:
I have just checked. This does work on Python 3.8, but does not work with Python 2.7. 

Anyway, it is irrelevant now, with the new version of this push request.

Reply:
> Actually I have not tried this, as I would never thought this can work.

Me too, I discovered it by accident, when working on this: https://github.com/spatialaudio/jackclient-python/blob/0151633118518a6b9bae406d1b66428b085e23f7/src/jack.py#L100-L122.

> This does work on Python 3.8, but does not work with Python 2.7.

OK, good to know.

I don't think the content of `args` is very important anyway, since we have separate attributes.